src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XDocument.cs

```diff
@@ -633,7 +633,7 @@ public async Task SaveAsync(Stream stream, SaveOptions options, CancellationToke
                 }
             }
 
-            using (XmlWriter w = XmlWriter.Create(stream, ws))
+            await using (XmlWriter w = XmlWriter.Create(stream, ws))
```

Comment:
We need to use ConfigureAwait(false) on the DisposeAsync calls.  The syntax isn't ideal; this will need to change to be more like:
```C#
XmlWriter w = XmlWriter.Create(stream, ws);
await using (w.ConfigureAwait(false))
```
Same goes for the other occurrences below.